core/chains/evm/txmgr/txmgr.go

```diff
@@ -248,20 +279,45 @@ func (b *Txm) runLoop(eb *EthBroadcaster, ec *EthConfirmer) {
 			eb.Trigger(address)
 		case head := <-b.chHeads:
 			ec.mb.Deliver(head)
+		case f := <-b.reset:
+			if err := eb.Close(); err != nil {
+				panic(err)
```

Comment:
Is panic our only option? Can we log panic with some context at least?
```suggestion
				b.logger.Panicw("Failed to Close EthBroadcaster", "error", err)
```

Reply:
It shouldn't ever happen, the function can't error unless it was already stopped, which is definitely a programming error.

Reply:
Maybe use`lggr.AssumptionViolation`?

Reply:
I want the application to exit if this happens, it should not be possible and indicates we got into an unknown state.